### ISSUE TITLE
Add a method to modify the deadline of the sessionData

### ISSUE THREAD

**Issue by lwj5** [alexedwards/scs#149]
At the moment every session has a fixed lifetime which is set on the initialization of the session manager. There is no way to modify the deadline for an individual session after.

It would be useful if a method can be exposed to set the expiry for cases when the deadline differs from the default or modified due to other conditions.

```golang
func (s *SessionManager) Expire(ctx context.Context, expiry time.Time) {
	sd := s.getSessionDataFromContext(ctx)

	sd.mu.Lock()
	defer sd.mu.Unlock()
	sd.deadline = expiry
	sd.status = Modified
}
```

Let me know if I should proceed with a PR

**Comment by anemajc** [comment]
We also need this functionality.

**Comment by alexedwards** [comment]
@lwj5 Thanks, yes I'd be happy for a PR. I'm not sure that the method name `Expire()` is ideal --- perhaps `SetDeadline()` would be more explicit and complement with the existing `Deadline()` method?

**Comment by lwj5** [comment]
Thanks for the reply. I'm maintaining a fork at https://github.com/Inquisico/go-session with an extended set of features we require. I'll add `SetDeadline` here when I have the time.

Feel free to do so if anyone else would like to go at it.

**Comment by skast96** [comment]
Has this been added via a PR or is this still open? I need that feature as well.

**Comment by alexedwards** [comment]
There's no PR yet, as far as I know. I haven't had time to work on it, so a PR would still be very much appreciated.

**Comment by petersanchez** [comment]
This is great. I was looking into how to modify scs to support just this. I see the PR #181 is already submitted! Excellent.

**Comment by alexedwards** [comment]
PR merged. `SetDeadline()` is now available in v2.7.0.
